lib/mongo_compat_api.js

```diff
@@ -0,0 +1,325 @@
+     * @returns Collection instance if no callback passed
+     */
+    this.collection = function (name, options, callback) {
+      assert.ok(name, "collection name required");
+
+      options = this.copyDefaultOptions(options || {});
+      options.type = name;
+
+      return ditcher.doGetCollectionInstance(options, callback);
+    };
+
+    /**
+     * Create an index on collection
+     * http://mongodb.github.io/node-mongodb-native/2.0/api/Db.html#createIndex
+     * @param name Name of the collection
+     * @param fieldOrSpec Field to use for the index
+     * @param options (optional) Options
+     * @param callback (optional) result callback
+     * @returns Promise if no callback passed
+     */
+    this.createIndex = function (name, fieldOrSpec, options, callback) {
+      assert.ok(name, "collection name required");
+      assert.ok(fieldOrSpec, "index spec required");
+
+      if(typeof options === "function") {
+        callback = options;
+        options = {};
+      }
+
+      options = this.copyDefaultOptions(options || {});
+      options.type = name;
+      options.index = fieldOrSpec;
+
+      if (callback) {
+        ditcher.doIndex(options, callback);
+      } else {
+        return new Promise(function (resolve, reject) {
+          ditcher.doIndex(options, function (err, result) {
+            if (err) {
+              return reject(err);
+            }
+
+            resolve(result);
+          });
+        });
+      }
+    };
+
+    /**
+     * Rename a collection
+     * http://mongodb.github.io/node-mongodb-native/2.0/api/Db.html#renameCollection
+     * @param fromCollection Current name
+     * @param toCollection New name
+     * @param options (optional) Options
+     * @param callback (optional) Result callback
+     */
+    this.renameCollection = function (fromCollection, toCollection, options, callback) {
+      assert.ok(fromCollection, "current collection name required");
+      assert.ok(toCollection, "new collection name required");
+
+      if(typeof options === "function") {
+        callback = options;
+        options = {};
+      }
+
+      options = this.copyDefaultOptions(options || {});
+      options.type = fromCollection;
+      options.toCollection = toCollection;
+
+      if (callback) {
+        ditcher.doRenameCollection(options, callback);
+      } else {
+        return new Promise(function (resolve, reject) {
+          ditcher.doRenameCollection(options, function (err, result) {
+            if (err) {
+              return reject(err);
+            }
+
+            resolve(result);
+          });
+        });
+      }
+    };
+  } else {
+    // The MongoDB compatible API for per-app databases. Ideally it would be enough to just
+    // return the `#ditcher.database.db` object here (the mongo handle). However this results
+    // in a `promiseLibrary not defined` error (with or without bluebird). So we are wrapping
+    // the fhmongodb calls here.
```

Comment:
I think it worth spending sometime to investigate the issue. Ideally we don't want to wrapper the mongodb module for apps that have standalone dbs.